libs/src/main/python/dlpx/virtualization/libs/libs.py

```diff
@@ -410,3 +416,30 @@ def _log_request(message, log_level):
 
     response = internal_libs.log(log_request)
     _handle_response(response)
+
+
+def retrieve_credentials(credentials_supplier):
+    """This is an internal wrapper around the Virtualization library's credentials retrieval API.
+    Given a supplier provided by Virtualization, retrieves the credentials from that supplier.
+
+    Args:
+        credentials_supplier (dict): Properties that make up a supplier of credentials.
+    Return:
+        Subclass of Credentials retrieved from supplier. Either a PasswordCredentials or a KeyPairCredentials
+        from dlpx.virtualization.common._common_classes.
+    """
+    from dlpx.virtualization._engine import libs as internal_libs
```

Comment:
Any reason this isn't a top-level import?  Is this `internal_libs` the same or different than the one used in the above function `_handle_response`?

Reply:
I followed what every other method in this file does. Some of them have this comment explaining why it's there:
```
    #
    # Since this import only resolves at runtime, we keep it in the function
    # scope to allow unit testing of this module.
    #
    from dlpx.virtualization._engine import libs as internal_libs
```